python/simulation.cpp

```diff
@@ -97,7 +97,21 @@ class simulation_shim {
     }
 
     arb::time_type run(arb::time_type tfinal, arb::time_type dt) {
-        return sim_->run(tfinal, dt);
+        arb::time_type tepoch = 0;
+        while (true) {
+            if (PyErr_CheckSignals() != 0) {
+                throw py::error_already_set();
+            }
+            tepoch += dt * 1000;
```

Comment:
Depending on the model 1000 $\Delta t$ can already be an appreciable delay. Not sure we should use a magic number here.

Reply:
Alternatively, we (different options):

 - call the check function each dt
 - check the elapsed time every 1dt and call the check function if that's more than 100ms
 - record the time it takes to execute the first dt, extrapolate how many steps it would take to execute (>) 100ms, and take that many steps in between.